ToolChain/Assembler/Constants.py

```diff
@@ -44,3 +44,40 @@
 
 UNDEFINED = "&&undefined&&"
 DEFAULT_OUTPUT_EXTENSION = ".o"
+
+REGISTERS = {
+    "A": 0b0000, "B": 0b0001, "C": 0b0010, "D": 0b0011,
+    "E": 0b0100, "F": 0b0101, "G": 0b0110, "S": 0b0111,
+
+    "A2": 0b1000, "B2": 0b1001, "C2": 0b1010, "D2": 0b1011,
+    "E2": 0b1100, "F2": 0b1101, "G2": 0b1110, "S2": 0b1111
+}
+
+INSTRUCTION_LIST = [
+    "ACTI", "ADD", "AND", "CALL", "CMP", "DACTI", "DIV",
+    "HIRET", "INT", "JMP", "JMPR", "MEMR", "MEMW", "MOV",
+    "MUL", "NOP", "NOT", "NOP", "OR", "POP", "PUSH",
+    "RET", "SFSTOR", "SIVR", "SHL", "SHR", "SUB", "XOR"
+]
+
+LABEL_INSTRUCTIONS = [
```

Comment:
There are more instructions that can be used with labels than these instruction. Looking at the parser code, the code in very similar. For example, the code require for the InsWidthImmReg if somewhat close to the MOV instruction code required to handle the label. Is there any reasons why this list is required? I think the code could be unified. The reason why I don't like this list here is:
-Leads code reader into thinking only these instructions can be used with labels
-Would require maintenance in the future
-Creates multiple ways of handling the labels in the parser
Every instruction that has the ability to use an immediate value technically is a good candidate to use a label.

Reply:
if every instruction that can handle an immediate can also handle labels, then this whole thing won't be necessary.